src/logic/safe/store/actions/fetchSafe.ts

```diff
@@ -1,178 +1,187 @@
-      abi: GnosisSafeSol.abi as AbiItem[],
-      address: safeAddress,
-      methods: safeParams,
-    }),
-    getSafeInfo(safeAddress),
-    getLocalSafe(safeAddress),
-    getBalanceInEtherOf(safeAddress),
-  ])
+const extractLocalSafeInfo = (
+  localSafeInfoSettled: PromiseSettledResult<SafeRecordProps | undefined>,
+  remoteSafeInfoRetrievedSuccessfully: boolean,
+): Partial<SafeRecordProps> => {
+  const safeInfo: Partial<SafeRecordProps> = {}
+
+  if (localSafeInfoSettled.status === 'fulfilled') {
+    if (localSafeInfoSettled.value !== undefined) {
+      const localSafeInfo = localSafeInfoSettled.value
+
+      if (!remoteSafeInfoRetrievedSuccessfully) {
+        // if for any reason we failed to retrieve safe's remote information,
+        // we default to what we had locally stored
+        safeInfo.nonce = localSafeInfo.nonce
+        safeInfo.threshold = localSafeInfo.threshold
+        safeInfo.currentVersion = localSafeInfo.currentVersion
+        safeInfo.needsUpdate = localSafeInfo.needsUpdate
+        safeInfo.featuresEnabled = localSafeInfo.featuresEnabled
+        safeInfo.spendingLimits = localSafeInfo.spendingLimits
+        safeInfo.modules = localSafeInfo.modules
+      }
+
+      safeInfo.name = localSafeInfo.name
+    }
+  } else {
+    console.error('Failed to load locally stored Safe information', localSafeInfoSettled.reason)
+  }
 
-  const threshold = Number(thresholdStr)
-  const nonce = Number(nonceStr)
-  const owners = buildOwnersFrom(remoteOwners, localSafe)
-  const needsUpdate = safeNeedsUpdate(currentVersion, latestMasterContractVersion)
-  const featuresEnabled = enabledFeatures(currentVersion)
-  const modules = await getModules(safeInfo)
-  const spendingLimits = safeInfo ? await getSpendingLimits(safeInfo.modules, safeAddress) : null
+  return safeInfo
+}
 
-  return {
-    address: safeAddress,
-    name: safeName,
-    threshold,
-    owners,
-    ethBalance,
-    totalFiatBalance: totalFiatBalance || 0,
-    nonce,
-    currentVersion: currentVersion ?? '',
-    needsUpdate,
-    featuresEnabled,
-    balances: localSafe?.balances || Map(),
-    latestIncomingTxBlock: 0,
-    activeTokens: Set(),
-    modules,
-    spendingLimits,
+const extractSafeOwners = (
+  remoteSafeInfoSettled: PromiseSettledResult<SafeInfo>,
+  localSafeInfoSettled: PromiseSettledResult<SafeRecordProps | undefined>,
+): List<SafeOwner> | undefined => {
+  let localOwners
+  let remoteOwners
+
+  if (localSafeInfoSettled.status === 'fulfilled' && localSafeInfoSettled.value !== undefined) {
+    const localSafeInfo = localSafeInfoSettled.value
+
+    localOwners = localSafeInfo.owners
   }
+
+  if (remoteSafeInfoSettled.status === 'fulfilled') {
+    const remoteSafeInfo = remoteSafeInfoSettled.value
+
+    remoteOwners = remoteSafeInfo.owners.map(({ value }) => {
+      const localOwner = localOwners?.find(({ address }) => sameAddress(address, value))
+      const name = localOwner?.name ?? 'UNKNOWN'
```

Comment:
@katspaugh In my understanding, it's a mix of privacy concerns and technical complexity. If we store them on the backend, they have to be either public or private. The public is not ideal,  it could expose someone's identity. If you make it private, you need to authorize the user to view them somehow which could be complex. Maybe you can raise this question with Richard

Reply:
Gotcha, thanks!
Still need to wrap my head around the everything-is-public concept.